---
format:
  revealjs:
    incremental: false
    theme: [default, custom.scss]
execute: 
  echo: false
---

```{r}
# label: setup
suppressPackageStartupMessages(library(tidyverse))
library(targets)
library(ape)
library(ggtree)

# Define location of targets cache from goflag filmies analysis
filmy_store <- "data/_targets-2025-02-26"

# Rescale a tree
rescale_tree_len <- function (tree, scale){
	tree$edge.length <- tree$edge.length / max(phytools::nodeHeights(tree)) * scale
	tree
}

# Create a bootstrap tibble for plotting tree
make_bs_tbl <- function(tree, cutoff = 0) {
  # Make a dataframe (tibble) with node IDs (integers) and their corresponding bootstrap support values
  # The tibble has two columns: one called "node", the other can be named as we like (here, "bootstrap")
  bs_tibble <- tibble(
    # hard-code node ID: internal nodes start after tip nodes,
    # and phy$node.label is in the same order as internal nodes
    node = 1:Nnode(tree) + Ntip(tree), 
    bootstrap = parse_number(tree$node.label)
    )
  if (cutoff > 0) {
    bs_tibble <-
    bs_tibble |>
      mutate(bootstrap = if_else(bootstrap < cutoff, "", bootstrap))
  }
  bs_tibble
}
```

## 次世代 DNA シーケンスによる<br>コケシノブ科の系統解析:<br>深い系統関係の解明 {background-color="black" background-image="images/IMG_6580.png" background-size="cover" background-repeat="no-repeat" background-position="center"}

:::: {.columns}

::: {.column width="50%"}

::: {.medsmall}

**ニッタ ジョエル(千葉大・国際教養)**・Weston Testo (Univ. Vermont)・Michael Sundue (Royal Botanic Garden Edinburgh)・Alejandra Vasco (Fort Worth Botanic Garden)・ Laura Kragh Frederiksen (Aarhus Univ.)・Hanna Tuomisto (Aarhus Univ. / Univ. Turku)・ GoFlag Consortium・海老原淳(科博・植物)

日本植物分類学　第24回大会

2025-03-10 OB14

:::


:::

::::


# 背景

## コケシノブ科

:::: {.columns}

::: {.column width="30%"}
- 一細胞層から成る葉を持つ独特なシダ植物系統
- やく600種
- 主に熱帯・多湿な環境
:::

::: {.column width="70%"}
![](images/filmies_examples.png)
:::

::::

## 二つの大きな系統

:::: {.columns}

::: {.column width="45%"}
### *Hymenophyllum*

![](images/involucre_bivalvate_ebihara.png){height="400"}

- 二弁状の包膜
:::

::: {.column width="10%"}
:::

::: {.column width="45%"}
### 広義の*Trichomanes*

![](images/involucre_tube_ebihara.png){height="400"}

- 筒状の包膜
:::

::::

## *Hymenophyllum*

:::: {.columns}

::: {.column width="50%"}
- 約300種
- 形態・生態的に**均一**
  - 着生
  - 中型
:::

::: {.column width="50%"}
![](images/hymenophyllum_examples.png){height="600"}
:::

::::

## 広義の*Trichomanes*

:::: {.columns}

::: {.column width="50%"}
- 約300種
- 形態、習性とも**多様**
  - 岩上生
  - 地上性
  - 半着生
  - 着生
  - 小型から大型まで
:::

::: {.column width="50%"}
![](images/trichomanoid_habit_examples.png){height="450"}
:::

::::

## 分類の再検討（Ebihara et al. 2006）

:::: {.columns}

::: {.column width="30%"}
- *Hymenophyllum*を10亜属に分ける
- 広義の*Trichomanes*を8属に分ける
:::

::: {.column width="70%"}
![](images/ebihara_2006_tree.png)
:::

::::

## 残っている疑問

:::: {.columns}

::: {.column width="50%"}
- 属間の関係が明らかになっていない
- どの順番で地上性から着生（あるいは、その反対）に進化したのか？
:::

::: {.column width="50%"}
![](images/dubuisson_2003_theory.png)

::: {.small}
Dubuisson, J.-Y., et al. 2003 Bot. J. Linn. 142, 41–63.

:::

:::

::::

## 本研究の目的

- **ゲノムワイドなデータセット**を用いてコケシノブ科の系統関係を明らかにする
- コケシノブ科における習性の**進化の再検討**を行う

# 方法

## サンプリング

- Ebihara et al. (2006)の属・亜属・セクションを網羅する

- その後に発表された*Trichomanes*の亜属が入っていない

---

::: {.medsmall}

```{r}
#| label: sampling
tar_load(combined_samples, store = filmy_store) 
tar_load(samples_with_subgen, store = filmy_store)

samples_with_subgen <-
  samples_with_subgen |>
  mutate(
   subgenus = case_when(
    species == "Didymoglossum fulgens" ~ "Microgonium",
    species == "Didymoglossum sublimbatum" ~ "Microgonium",
    .default = subgenus
   ) 
  )

samples_for_table <-
  combined_samples |>
  select(sample_code) |>
  left_join(
    samples_with_subgen, by = join_by(sample_code)
  ) |>
  # Fill in missing subgenera
  mutate(
    subgenus = case_when(
      genus == "Hymenophyllum" & specific_epithet == "nephrophyllum" ~ "Cardiomanes",
      genus == "Hymenophyllum" & specific_epithet == "tegularis" ~ "Sphaerocionium",
      genus == "Vandenboschia" & specific_epithet == "striata" ~ "Vandenboschia",
      .default = subgenus
    )
  )


taxon_count <- 
  samples_for_table |>
  group_by(genus, subgenus, section) |>
  count() |>
  ungroup() |>
  arrange(genus, subgenus, section) |>
  mutate(across(everything(), ~replace_na(., "")))

n_samples_total <- sum(taxon_count$n)

taxon_count |>
  filter(genus != "Hymenophyllum") |>
  knitr::kable()
```

:::

---

::: {.medium}

```{r}
#| label: sampling-hym
taxon_count |>
  filter(genus == "Hymenophyllum") |>
  knitr::kable()
```

合計 `r n_samples_total` サンプル

:::

## GoFLAG (Geneology of Flagellate Plants)

:::: {.columns}

::: {.column width="40%"}
- 鞭毛付き植物の系統プロジェクト
  - 被子植物**以外**の陸上植物の全種を対象
  - やく400個の核マーカーを含む
:::

::: {.column width="60%"}
![](images/goflag-paper.png)

![](images/cycad_sperm.png){height=300}
:::

::::

## シーケンスキャプチャー法

![](images/seq-cap.png)

::: {.aside}
::: {.small}
<https://www.pacb.com/wp-content/uploads/PCR-based-capture.svg>
:::
:::

## 核マーカーのシーケンス

- CTAB法によるDNA抽出
- RAPiD Genomics社あるいはNovogene社に外注
  - GoFLAGマーカーによるシーケンスキャプチャー
  - 会社によって異なるプラットフォーム
    - NovaSeq X Plus (150bp PE)
    - HiSeq 3000 (100bp PE)

## 核マーカーの情報処理 {.smaller}

:::: {.columns}

::: {.column width="60%"}
- `fastp`でリードをクリーニング
- `HybPiper`パイプラインによるシーケンスエンリッチメント
  - `bwa`でリードをリファレンスにマッピングし、遺伝子ごとに分類
  - `SPAdes`でコンティグをde novoアセンブル
  - `Exonerate`でコーディング領域を抽出
:::

::: {.column width="40%"}
![](images/hybpiper_flow.png)

:::

::::

## 「対象外リード」の活用

![](images/seq-cap-off-target.png)

## 葉緑体(対象外)マーカーの情報処理 {.smaller}

:::: {.columns}

::: {.column width="50%"}
- `fastp`でリードをクリーニング
- `HybPiper`パイプラインの一部を使用
  - `bwa`でリードをリファレンスにマッピングし、遺伝子ごとに分類
- `bbmap`でリードをリファレンスに再マッピング
- `kindel`でコンセンサス配列を抽出
  - 最低カバレッジを1に設定
:::

::: {.column width="50%"}
![](images/kindel_flow.png)

::: {.small}
Constantinides, B., Robertson, D.L., 2017. JOSS 2, 282. <https://doi.org/10/gjzwjz>
:::

:::

::::

## 葉緑体(対象外)マーカーの情報処理

```{r}
#| label: plastid-gene-sampling
tar_load(c(
  combined_aligned_trimmed_plastome_tbl, combined_aligned_trimmed_sanger_tbl,
  combined_plastome_aln, combined_sanger_aln), store = filmy_store) 

n_plastome_taxa <- nrow(combined_plastome_aln)
n_plastome_markers <- n_distinct(combined_aligned_trimmed_plastome_tbl$locus)
n_plastome_bp <- ncol(combined_plastome_aln) |> scales::number(big.mark = ",")

n_sanger_taxa <- nrow(combined_sanger_aln)
n_sanger_markers <- n_distinct(combined_aligned_trimmed_sanger_tbl$locus)
n_sanger_bp <- ncol(combined_sanger_aln) |> scales::number(big.mark = ",")
```

- 新しく得た葉緑体データをFern Tree of Life v1.7.0（Nitta et al. 2022）のデータと組み合わせる
- 二つの葉緑体データセットを用意する：
  - 「全葉緑体」(`r n_plastome_taxa` 種 x `r n_plastome_markers` 遺伝子、`r n_plastome_bp` bp)
  - 「サンガー法」(`r n_sanger_taxa` 種 x `r n_sanger_markers` 遺伝子、`r n_sanger_bp` bp)

## 系統解析

- `iqtree`で最尤法
  - 1000回ブートストラップ
- 適切な外群がないため、*Hymenophyllum*と*Trichomanes*（広義）の間にルートを置く

---

葉緑体マーカーの場合は全葉緑体データの系統樹を束縛にサンガー法データセットで系統樹を作成

![](images/plastid_tree_flow.png){height="300"}

Nitta et al. (2022)

## 祖先形質復元

- `phytools`で三つのモーデルを作成
  - `ER` (Equal Rates): すべての状態間の遷移率が等しい
  - `SYM` (Symmetric): 状態の変化は対称的で、遷移率は異なるが逆方向の変化は同じ
  - `ARD` (All Rates Different): すべての遷移率が異なる
- AICで最もデータに合うモデルを選択し、祖先形質復元を行う

# 結果と考察

---

### リードの１割が対象遺伝子にマップされた

```{r}
tar_load(hybpiper_gene_stats_list, store = filmy_store)
tar_load(hybpiper_gene_stats_list_plastome, store = filmy_store)

read_stats_nuc <-
  hybpiper_gene_stats_list[[2]] |>
  select(
    name = Name, num_reads = NumReads, reads_mapped = ReadsMapped,
    perc_on_target = PctOnTarget) |>
  mutate(type = "nuclear")

read_stats_plastome <-
  hybpiper_gene_stats_list_plastome[[2]] |>
  select(
    name = Name, num_reads = NumReads, reads_mapped = ReadsMapped,
    perc_on_target = PctOnTarget) |>
  mutate(type = "plastome")

invisible(
  assertthat::assert_that(
    n_distinct(read_stats_nuc$name) == n_distinct(read_stats_plastome$name),
    msg = "Same number of samples not detected in nuclear and plastome data"
  )
)

n_samples <- n_distinct(read_stats_nuc$name)

read_stats <- read_stats_nuc |>
  bind_rows(read_stats_plastome)

read_stats |>
  group_by(type) |>
  summarize(
    mean_num_reads = mean(num_reads),
    mean_reads_mapped = mean(reads_mapped),
    mean_perc_on_target = mean(perc_on_target),
    sd_num_reads = sd(num_reads),
    sd_reads_mapped = sd(reads_mapped),
    sd_perc_on_target = sd(perc_on_target)
  ) |>
  mutate(
    across(
      contains("perc_"),
      ~scales::percent(. * 0.01, accuracy = 0.01)
    ),
    across(
      contains("reads"),
      ~scales::scientific(., digits = 2)
    )
  ) |>
  mutate(
    type = type,
    num_reads = glue::glue(
      "{mean_num_reads} ± {sd_num_reads}"
    ),
    mapped_reads = glue::glue(
      "{mean_reads_mapped} ± {sd_reads_mapped}"
    ),
    ,
    perc_on_target = glue::glue(
      "{mean_perc_on_target} ± {sd_perc_on_target}"
    ),
    .keep = "none"
  ) |>
  knitr::kable()
```

*n* = `r n_samples`

```{r}
#| label: rec-plot-prep

tar_load(
  c(hybpiper_rec_lens, hybpiper_rec_lens_plastome),
  store = "data/_targets-2025-02-26")

n_markers_nuc <- n_distinct(hybpiper_rec_lens$gene)

n_markers_plastome <- n_distinct(hybpiper_rec_lens_plastome$gene)

mean_rec_len_nuc <- hybpiper_rec_lens$perc_length |>
  mean() |>
  scales::percent(accuracy = 0.1)

mean_rec_len_plastome <- hybpiper_rec_lens_plastome$perc_length |>
  mean() |>
  scales::percent(accuracy = 0.1)
```

---

### 核遺伝子は平均 `r mean_rec_len_nuc` の解読率

```{r}
ggplot(hybpiper_rec_lens, aes(x = gene, y = species, fill = perc_length)) +
  geom_tile() +
  scale_fill_viridis_c() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank()
  )
```

---

### 葉緑体伝子（対象外）でも `r mean_rec_len_plastome`解読

```{r}
ggplot(hybpiper_rec_lens_plastome, aes(x = gene, y = species, fill = perc_length)) +
  geom_tile() +
  scale_fill_viridis_c() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank()
  )
```

<!-- Generate trees -->

```{r}
#| label: overall-tree
#| fig-height: 8
# Load tree
tar_load(
  c(hybpiper_dna_tree,
    samples_with_subgen),
  store = filmy_store)

hybpiper_dna_tree_plotting <- hybpiper_dna_tree

hybpiper_dna_tree_data <- tibble(
  tip = hybpiper_dna_tree$tip.label
) |>
  left_join(samples_with_subgen, relationship = "one-to-one")

tips_hym <- hybpiper_dna_tree_data |>
  filter(genus == "Hymenophyllum") |>
  pull(tip)

hybpiper_dna_tree_plotting <- 
  hybpiper_dna_tree_plotting |>
  root(tips_hym) |>
  rescale_tree_len(100)

# Prepare bootstrap tbl
bs_tibble <- make_bs_tbl(hybpiper_dna_tree_plotting)

dna_tree_plot <- ggtree(hybpiper_dna_tree_plotting) %<+%
  bs_tibble %<+%
  hybpiper_dna_tree_data +
  # add extra horizontal space for species names
  xlim(0, 120)

overall_tree <- dna_tree_plot +
  geom_tiplab(aes(label = species), size = 3) +
  geom_nodelab(aes(label = bootstrap), hjust = -0.1, size = 3) +
  annotate(
    "text", x = 0, y = Inf,
    label = glue::glue("GoFLAG マーカー（{n_markers_nuc}個）"),
    size = 6, hjust = 0.1, vjust = 1,
    family = "HiraKakuPro-W3") +
  annotate(
    "text", x = Inf, y = 0,
    label = "iqtreeによるML法\n1000回ブートストラップ",
    size = 6, hjust = 1, vjust = -0.5,
    family = "HiraKakuPro-W3")

ggsave(plot = overall_tree, file = "images/overall_tree.png", height = 11, width = 12)
```


```{r}
#| label: tric-tree
#| fig-height: 8

trich_nuc_tree <- hybpiper_dna_tree

trich_nuc_data <- tibble(
  tip = trich_nuc_tree$tip.label
) |>
  left_join(samples_with_subgen, relationship = "one-to-one", by = "tip")

tips_hym <- trich_nuc_data |>
  filter(genus == "Hymenophyllum") |>
  pull(tip)

trich_nuc_tree <- 
  trich_nuc_tree |>
  root(tips_hym) |>
  drop.tip(tips_hym) |>
  rescale_tree_len(100)

trich_nuc_data <- tibble(
  tip = trich_nuc_tree$tip.label
) |>
  left_join(samples_with_subgen, relationship = "one-to-one", by = "tip")

bs_tibble <- make_bs_tbl(trich_nuc_tree)

genus_labs <- trich_nuc_data |>
  group_by(genus) |>
  summarize(mrca = getMRCA(trich_nuc_tree, tip)) |>
  select(mrca, genus)

subgenus_labs <- trich_nuc_data |>
  filter(!is.na(subgenus)) |>
  add_count(subgenus) |>
  filter(n > 1) |>
  group_by(subgenus) |>
  summarize(mrca = getMRCA(trich_nuc_tree, tip)) |>
  select(mrca, subgenus)

trich_nuc_tree_plot <- ggtree(trich_nuc_tree, branch.length = "none") %<+%
  bs_tibble %<+%
  select(trich_nuc_data, tip, species) +
  # add extra horizontal space for species names
  xlim(0, 20)

trich_tree <- trich_nuc_tree_plot +
  geom_tiplab(aes(label = species), size = 5) +
  geom_nodelab(aes(label = bootstrap), hjust = -0.1, size = 3) +
  geom_cladelab(
    data = genus_labs,
    mapping = aes(node = mrca, label = genus),
    offset = 6,
    hjust = -0.1,
    extend = 0.25
    ) +
  geom_cladelab(
    data = subgenus_labs,
    mapping = aes(node = mrca, label = subgenus),
    offset = 10,
    hjust = -0.1,
    extend = 0.25
    ) +
  annotate(
    "text", x = 0, y = Inf,
    label = glue::glue("GoFLAG マーカー（{n_markers_nuc}個）"),
    size = 6, hjust = 0.1, vjust = 1,
    family = "HiraKakuPro-W3") +
  annotate(
    "text", x = Inf, y = 0,
    label = "iqtreeによるML法\n1000回ブートストラップ",
    size = 4, hjust = 1, vjust = -0.5,
    family = "HiraKakuPro-W3")

ggsave(plot = trich_tree, file = "images/trich_tree.png", height = 11, width = 12)
```

```{r}
#| label: tric-plastid-tree
#| fig-height: 8
#| fig-width: 12
tar_load(combined_sanger_tree, store = filmy_store)

trich_plastid_tree <- combined_sanger_tree

trich_plastid_data <- tibble(
  tip = trich_plastid_tree$tip.label
) |>
  left_join(samples_with_subgen, relationship = "one-to-one", by = "tip")

tips_hym <- trich_plastid_data |>
  filter(genus == "Hymenophyllum") |>
  pull(tip)

trich_plastid_tree <- 
  trich_plastid_tree |>
  root(tips_hym) |>
  drop.tip(tips_hym) |>
  # Remove single poorly placed Vandenboschia
  drop.tip("P044_WF02") |>
  rescale_tree_len(100)

bs_tibble <- make_bs_tbl(trich_plastid_tree)

trich_plastid_data <- tibble(
  tip = trich_plastid_tree$tip.label
) |>
  left_join(samples_with_subgen, relationship = "one-to-one", by = "tip")

genus_labs <- trich_plastid_data |>
  group_by(genus) |>
  summarize(mrca = getMRCA(trich_plastid_tree, tip)) |>
  select(mrca, genus)

subgenus_labs <- trich_plastid_data |>
  filter(!is.na(subgenus)) |>
  add_count(subgenus) |>
  filter(n > 1) |>
  group_by(subgenus) |>
  summarize(mrca = getMRCA(trich_plastid_tree, tip)) |>
  select(mrca, subgenus)

trich_plastid_tree_base <- ggtree(trich_plastid_tree, branch.length = "none") %<+%
  bs_tibble %<+%
  select(trich_plastid_data, tip, species) +
  # add extra horizontal space for species names
  xlim(0, 35)

subgenus_singleton <-
  tibble(
    mrca = MRCA(trich_plastid_tree_base, "Callistopteris_superba"),
    subgenus = "Asiatrichomanes"
  )

subgenus_labs <- subgenus_labs |>
  bind_rows(subgenus_singleton)

trich_plastid_tree_plot <-
  trich_plastid_tree_base +
  geom_tiplab(aes(label = species), size = 1.5) +
  geom_nodelab(aes(label = bootstrap), hjust = -0.1, size = 1.5) +
  geom_cladelab(
    data = genus_labs,
    mapping = aes(node = mrca, label = genus),
    offset = 6,
    hjust = -0.1,
    extend = 0.25
    ) +
  geom_cladelab(
    data = subgenus_labs,
    mapping = aes(node = mrca, label = subgenus),
    offset = 11,
    hjust = -0.1,
    extend = 0.25
    ) +
  annotate(
    "text", x = 0, y = Inf,
    label = glue::glue("葉緑体マーカー（{n_markers_plastome}個）"),
    size = 6, hjust = 0.1, vjust = 1,
    family = "HiraKakuPro-W3")

ggsave(plot = trich_plastid_tree_plot, file = "images/trich_plastid_tree.png", height = 11, width = 12)
```

```{r}
#| label: tric-ss-plastid-tree-zoom
#| fig-height: 8
trich_ss_plastid_tree <- 
  viewClade(trich_plastid_tree_base, MRCA(trich_plastid_tree_plot, "P044_WA03", "P044_WD04")) +
  geom_tiplab(aes(label = species), size = 3) +
  geom_nodelab(aes(label = bootstrap), hjust = -0.1, size = 3) +
  geom_cladelab(
    data = subgenus_labs,
    mapping = aes(node = mrca, label = subgenus),
    offset = 6,
    hjust = -0.1,
    extend = 0.25
    ) +
  annotate(
    "text", x = 0, y = Inf,
    label = glue::glue("葉緑体マーカー（{n_markers_plastome}個）"),
    size = 6, hjust = -0.2, vjust = 1,
    family = "HiraKakuPro-W3")

ggsave(plot = trich_ss_plastid_tree, file = "images/trich_ss_plastid_tree.png", height = 11, width = 12)
```

## {background-color="white" background-image="images/overall_tree.png" background-size="contain" background-repeat="no-repeat" background-position="center"}

## {background-color="white" background-image="images/trich_tree.png" background-size="contain" background-repeat="no-repeat" background-position="center"}

## {background-color="white" background-image="images/trich_plastid_tree.png" background-size="contain" background-repeat="no-repeat" background-position="center"}

## {background-color="white" background-image="images/trich_ss_plastid_tree.png" background-size="contain" background-repeat="no-repeat" background-position="center"}

## {background-color="white" background-image="images/anc_res_plot.png" background-size="contain" background-repeat="no-repeat" background-position="75% 100%"}

祖先形質復元